lib/facter/resolvers/aix/ffi/ffi_helper.rb

```diff
@@ -1,34 +1,152 @@
 # frozen_string_literal: true
 
 require 'ffi'
-
+require_relative 'structs'
+require_relative 'ffi'
 module Facter
-  module Aix
-    module FfiHelper
-      KINFO_GET_AVENRUN = 1
-      KINFO_READ        = 8 << 8
+  module Resolvers
+    module Aix
+      module FfiHelper
+        KINFO_GET_AVENRUN = 1
+        KINFO_READ        = 8 << 8
+        KINFO_RT          = 1 << 8
+        KINFO_RT_IFLIST   = KINFO_RT | 3
 
-      module Libc
-        extend FFI::Library
+        module Libc
+          extend ::FFI::Library
 
-        RTLD_LAZY   = 0x00000004
-        RTLD_GLOBAL = 0x00010000
-        RTLD_MEMBER = 0x00040000
+          RTLD_LAZY   = 0x00000004
+          RTLD_GLOBAL = 0x00010000
+          RTLD_MEMBER = 0x00040000
 
-        @ffi_lib_flags = RTLD_LAZY | RTLD_GLOBAL | RTLD_MEMBER
-        ffi_lib 'libc.a(shr.o)'
+          @ffi_lib_flags = RTLD_LAZY | RTLD_GLOBAL | RTLD_MEMBER
+          ffi_lib 'libc.a(shr.o)'
 
-        attach_function :getkerninfo, %i[int pointer pointer int], :int
-      end
+          attach_function :getkerninfo, %i[int pointer pointer int], :int
+          attach_function :inet_ntop, %i[int pointer pointer uint], :string
+        end
+
+        def self.read_load_averages
+          averages = ::FFI::MemoryPointer.new(:long_long, 3)
+          averages_size = ::FFI::MemoryPointer.new(:int, 1)
+          averages_size.write_int(averages.size)
+
+          return if Libc.getkerninfo(KINFO_READ | KINFO_GET_AVENRUN, averages, averages_size, 0).negative?
+
+          averages.read_array_of_long_long(3).map { |x| (x / 65_536.0).round(5) }
+        end
+
+        def self.read_interfaces
+          ksize = Libc.getkerninfo(KINFO_RT_IFLIST, nil, nil, 0)
+
+          ksize_ptr = ::FFI::MemoryPointer.new(:int, ksize.size)
+          ksize_ptr.write_int(ksize)
+
+          count = ksize / 1
+          result_ptr = ::FFI::MemoryPointer.new(:char, count)
+          res = Libc.getkerninfo(KINFO_RT_IFLIST, result_ptr, ksize_ptr, 0)
+
+          if res == -1
+          end
+          cursor = 0
+
+          interfaces = {}
+          while cursor < ksize_ptr.read_int
+            hdr = FFI::IfMsghdr.new(result_ptr + cursor)
+
+            case hdr[:ifm_type]
+            when FFI::RTM_IFINFO
+              link_addr = FFI::SockaddrDl.new(hdr.to_ptr + hdr.size)
+
+              interface_name = link_addr[:sdl_data].to_s[0, link_addr[:sdl_nlen]]
+              interfaces[interface_name] ||= {}
+
+            when FFI::RTM_NEWADDR
+              @addresses = {}
```

Comment:
does this have to be an instance variable?

Reply:
No, I'll update.